vispy/visuals/axis.py

```diff
@@ -0,0 +1,93 @@
+
+import numpy as np
+
+from .visual import Visual
+from .line import LineVisual
+from .text import TextVisual
+
+class AxisVisual(Visual):
+    """Axis visual
+
+    Parameters
+    ----------
+    extents : array
+        Co-ordinates of start and end of the axis.
+    tick_direction : array
+        Unit vector specifying the direction of the ticks. Default is
+        left/down. (This will be normalised to a unit vector)
+    domain : tuple
+        The data values at the beginning and end of the axis, used for tick
+        labels. i.e. (5, 10) means the axis starts at 5 and ends at 10.
+    scale_type : str
+        The type of scale. Default is 'linear'. Options:
+            * "linear"
+            * "logarithmic"
+            * "power"
+    """
+    def __init__(self, extents, tick_direction = None, domain=(0., 1.),
+                 scale_type="linear", axis_color=(1, 1, 1, 1),
+                 tick_color = (0.7, 0.7, 0.7, 1)):
+        self.axis_color = axis_color
+        self.tick_color = tick_color
+
+        self.extents = extents
+        self.vec = np.subtract(self.extents[1],self.extents[0])
+
+        self.tick_direction = tick_direction or self._get_tick_direction()
+
+    def draw(self, transforms):
+        v_line = LineVisual(pos=self.extents, color=self.axis_color,
+                          mode='gl', width=3.0)
+
+        tick_fractions, tick_labels = self._get_tick_frac_labels()
+
+        tick_pos, tick_label_pos = self._get_tick_positions(tick_fractions)
+
+        v_ticks = LineVisual(pos=tick_pos, color=self.tick_color, mode='gl',
+                           width=2.0, connect='segments')
+
+        # TODO: make this a more efficient iterator
+        for x in range(0, len(tick_labels)):
+            TextVisual(str(tick_labels[x]), font_size=8, color='w',
+                       pos=tick_label_pos[x]).draw(transforms)
```

Comment:
I completely agree that is what we should do eventually. But there is some
utility in continuing down the current path. Right now the collections
cannot be used directly as visuals, so that would need to be done before
any progress could be made here. In the meantime, creating the visuals once
should give at least decent enough performance that we could at least
address some high-level questions. For instance, if it at least works okay,
then I could start integrating it in my plotting PR within a week, instead
of having to wait what I assume would be a month or more before the
collections-as-visuals work well, and then get implemented here...

@nippoo if you want to try to take over or work with @campagnola on making
the collections proper visuals that integrate properly with the scenegraph
that would be great, but in my mind it really depends on how much time you
have to devote to this stuff.